File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans = []
        def backtrack(li ,start):
            nonlocal ans , nums
            ans.append(li)
            for i in range(start , len(nums)):
                backtrack(li+[nums[i]] , i+1)
        backtrack([] , 0 )
        return ans

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> ans;
    void traverse(vector<int> li , int start , vector<int> &nums ){
        if (start > nums.size()) return ;
        ans.push_back(li);
        for (int i = start ; i<nums.size() ; i++){
            li.push_back(nums[i]);
            traverse(li , i +1 , nums);
            li.pop_back();
        }
        return ;
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        traverse({} , 0 , nums);
        return ans;
    }
};